addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -109,18 +139,36 @@ private void addApplicationsToSite(String site, ApplicationMatch applicationMatc
 
     private ApplicationMatch checkAppMatches(
             ApplicationMatch appMatch, Application currentApp, HttpMessage msg, Source source) {
-        appMatch = checkUrlMatches(appMatch, currentApp, msg);
-        appMatch = checkHeadersMatches(appMatch, currentApp, msg);
-        appMatch = checkCookieMatches(appMatch, currentApp, msg);
+        // Process headers to identify techs
+        for (CustomMessageProcessor cmp : messageHeaderProcessors) {
+            appMatch = cmp.process(appMatch, currentApp, msg);
+            if (!exhaustive && appMatch != null) {
+                return appMatch;
+            }
+        }
+
         if (!msg.getResponseHeader().isText()) {
             return appMatch; // Don't check body if not text'ish
         }
+
+        // Process body to identify techs - This is ordered to hopefully match earlier
```

Comment:
Aren't they all ordered that way, it would be better if that was the case.

Reply:
Yes they are, I do t know why I felt the need to make this comment different. But, you're right they're kinda overkill.